src/FunctionInvocationBinary.cpp

```diff
@@ -370,9 +410,35 @@ FunctionInvocationBinary::Output(std::ostream &out) const
 				op_flags->OutputSize(out);
 				out << ")";
 			}
+
+			//float_test
+			bool should_use_float_macro = false;
+			if (CGOptions::float_test() &&
+					(param_value[0]->get_type().is_float() || param_value[1]->get_type().is_float())){
+				should_use_float_macro = true;
+				output_func_float_macro(eFunc, out);
+			}
+
+			bool should_close_brackets = false;
+			if (CGOptions::float_test() && should_use_float_macro
+					&& !param_value[0]->get_type().is_float()){
+				output_cast_to_interval_macro(out, param_value[0]->get_type());
+				should_close_brackets = true;
+			}
+
 			param_value[0]->Output(out);
+
+			if (CGOptions::float_test() && should_close_brackets) {
+				out<< ")";
+			}
+
 			out << " ";
-			OutputStandardFuncName(eFunc, out);
+			if(should_use_float_macro){
```

Comment:
Similar to should_close_brackets, can we also add ```CGOptions::float_test()```? I know it's a bit redundant, but it might be safer (and for documentation purpose as well).